### ISSUE TITLE
Error importing AKS cluster with Terraform rancher2_cluster resource (aks_config_v2)

### ISSUE THREAD

**Issue by mazlumtoprak** [rancher/terraform-provider-rancher2#757]
Hey all,

We want to to import an `azurerm` created cluster with a default node pool wich has autoscaling enabled into rancher with the new version and the `aks_config_v2`. The default node pool also has been created with the azurerm provider.

When we manually import the cluster via Rancher2 UI, we have no issues, everything looks fine. While using the terraform provider, we get some errors. This is what our `rancher2_cluster` config looks like.

**Our Setup**

- Rancher = v2.6.0 
- Terraform Rancher2 provider version = "1.20.0"
- AKS Kubernetes Version = 1.20.7

```
resource "rancher2_cloud_credential" "aks" {
  name = "aks-cloud-credential"
  azure_credential_config {
    client_id       = var.clientid
    client_secret   = var.clientsecret
    subscription_id = var.subscriptionid
  }
}

resource "rancher2_cluster" "aks-import" {
  name        = "aks-imported-cluster"
  description = "Terraform AKS imported cluster"
  aks_config_v2 {
    name                = azurerm_kubernetes_cluster.aks.name
    cloud_credential_id = rancher2_cloud_credential.aks.id
    resource_group      = azurerm_resource_group.aks.name
    resource_location   = azurerm_resource_group.aks.location
    imported            = true
  }
}
```
We have observed the following error:

```
Waiting for API to be available:failed to remove node pool: containerservice.AgentPoolsClient#Delete: Failure sending request: StatusCode=0 -- Original Error: Code="OperationNotAllowed" Message="There has to be at least one agent pool."
```

Not sure, why it can't fetch all the details from the cluster, that we've specified above. When we add the node_pools block with a reference to the default_node_pool name from the azurerm resource, we are probably overwriting the already created node_pool and get hints, that we need to define the full node_pool again, hence duplicating it.

```
| Failed to update cluster:  containerservice.AgentPoolsClient#CreateOrUpdate: Failure sending  request: StatusCode=0 -- Original Error: Code="PropertyChangeNotAllowed"  Message="Changing property 'properties.maxPods' is not allowed."  Target="properties.maxPods"
-- | --
```

The method with defining the node_pool on our own in the aks_config_v2, it modifies the resources on azure.

Are we having a false assumption, that we can import an existing cluster with this resource into rancher? What are we missing?

Glad to provide you more details if needed. Thanks!

**Comment by rawmind0** [comment]
Hey @mazlumtoprak , issue reproduced. Submitted PR  https://github.com/rancher/terraform-provider-rancher2/pull/775 to fix it.

**Comment by rawmind0** [comment]
PR #775 is already merged. The fix will be available at next tf provider release, v1.21.0
